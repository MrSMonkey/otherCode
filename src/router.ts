import Vue from 'vue';
import Router from 'vue-router';
import Home from '@/views/Home/Home.vue';
Vue.use(Router);

let router!: any;

const Bind = () => import(/* webpackChunkName: 'bind' */ '@/views/Login/Bind.vue');

const Entrust = () => import(/* webpackChunkName: 'entrust' */ '@/views/House/Entrust.vue');
const Perfect = () => import(/* webpackChunkName: 'perfect' */ '@/views/House/Perfect.vue');
const House = () => import(/* webpackChunkName: 'house' */ '@/views/House/House.vue');
const EntrustPlan = () => import(/* webpackChunkName: 'entrustPlan' */ '@/views/House/EntrustPlan.vue');
const ChoiceHouse = () => import(/* webpackChunkName: 'choiceHouse' */ '@/views/House/ChoiceHouse.vue');

const MyHouse = () => import(/* webpackChunkName: 'myHouse' */ '@/views/HouseDetail/MyHouse.vue');
const HouseImages = () => import(/* webpackChunkName: 'houseImages' */ '@/views/HouseDetail/HouseImages.vue');

const ServiceOrder = () => import(/* webpackChunkName: 'serviceOrder' */ '@/views/ServiceOrder/ServiceOrder.vue');
const ServiceDetile = () => import(/* webpackChunkName: 'serviceDetile' */ '@/views/ServiceOrder/ServiceDetile.vue');
const ProductDetile = () => import(/* webpackChunkName: 'productDetile' */ '@/views/ServiceOrder/ProductDetile.vue');
const ServiceOrderDetail = () => import(/* webpackChunkName: 'serviceOrderDetail' */ '@/views/ServiceOrder/ServiceOrderDetail.vue');
const ServiceInfo = () => import(/* webpackChunkName: 'serviceInfo' */ '@/views/ServiceOrder/ServiceInfo.vue');
const ProductInfo = () => import(/* webpackChunkName: 'productInfo' */ '@/views/ServiceOrder/ProductInfo.vue');

const Purchase = () => import(/* webpackChunkName: 'purchase' */ '@/views/Service/Purchase.vue');
const ServiceType = () => import(/* webpackChunkName: 'serviceType' */ '@/views/Service/ServiceType.vue');
const StartService = () => import(/* webpackChunkName: 'startService' */ '@/views/Service/StartService.vue');
const NotFoundComp = () => import(/* webpackChunkName: '404' */ '@/components/404.vue');

const MaintainChecked = () => import(/* webpackChunkName: 'maintainChecked' */ '@/views/ServiceRecord/MaintainChecked.vue');
const ServiceRecord = () => import(/* webpackChunkName: 'serviceRecord' */ '@/views/ServiceRecord/ServiceRecord.vue');
const ConfirmPay = () => import(/* webpackChunkName: 'confirmPay' */ '@/views/ServiceRecord/ConfirmPay.vue');
const ServiceRecordLook = () => import(/* webpackChunkName: 'serviceRecordLook' */ '@/views/ServiceRecord/ServiceRecordLook.vue');

const HouseAppraise = () => import(/* webpackChunkName: 'houseAppraise' */ '@/views/HouseAppraise/HouseAppraise.vue');
const AppraiseHouseInfo = () => import(/* webpackChunkName: 'houseAppraise' */ '@/views/HouseAppraise/AppraiseHouseInfo.vue');

router = new Router({
  base: process.env.BASE_URL,
  routes: [
    { path: '/', name: 'index', component: Home, meta: '星空业主服务号' },
    { path: '/entrustPlan', name: 'entrustPlan', meta: '星级房屋托管计划', component: EntrustPlan},
    { path: '/bind', name: 'bind', meta: '登录', component: Bind},
    { path: '/entrust', name: 'entrust', meta: '在线委托', component: Entrust},
    { path: '/perfect', name: 'perfect', meta: '完善房源信息', component: Perfect},
    { path: '/house', name: 'house', meta: '我的房源列表', component: House},
    { path: '/choiceHouse', name: 'choiceHouse', meta: '选择房源', component: ChoiceHouse},
    { path: '/myHouse', name: 'myHouse', meta: '房源详情', component: MyHouse},
    { path: '/serviceOrder', name: 'serviceOrder', meta: '服务订单', component: ServiceOrder},
    { path: '/serviceDetile', name: 'serviceDetile', meta: '订单详情', component: ServiceDetile},
    { path: '/productDetile', name: 'productDetile', meta: '订单详情', component: ProductDetile},
    { path: '/purchase', name: 'purchase', meta: '购买服务包', component: Purchase},
    { path: '/serviceInfo', name: 'serviceInfo', meta: '服务包详情', component: ServiceInfo},
    { path: '/productInfo', name: 'productInfo', meta: '服务产品详情', component: ProductInfo},
    { path: '/serviceType', name: 'serviceType', meta: '选择服务类型', component: ServiceType},
    { path: '/startService', name: 'startService', meta: '发起服务', component: StartService},
    { path: '/404', name: '404', meta: '404', component: NotFoundComp},
    { path: '/maintainChecked', name: 'maintainChecked', meta: '确认验收', component: MaintainChecked},
    { path: '/serviceRecord', name: 'serviceRecord', meta: '服务记录', component: ServiceRecord},
    { path: '/confirmPay', name: 'confirmPay', meta: '确认支付', component: ConfirmPay},
    { path: '/serviceRecordLook', name: 'serviceRecordLook', meta: '服务记录', component: ServiceRecordLook},
    { path: '/serviceOrderDetail', name: 'serviceOrderDetail', meta: '服务详情', component: ServiceOrderDetail},
    { path: '/serviceOrderDetail', name: 'serviceOrderDetail', meta: '服务详情', component: ServiceOrderDetail},
    { path: '/houseImages', name: 'houseImages', meta: '房源照片', component: HouseImages},
    {
      path: '/houseAppraise',
      name: 'houseAppraise',
      meta: '房屋估价',
      component: HouseAppraise,
      children: [
        {path: 'appraiseHouseInfo', name: 'appraiseHouseInfo', meta: '估价房屋信息', component: AppraiseHouseInfo }
      ]
    }
  ]
});

router.beforeEach((to: any, from: any, next: any) => {
  /* 路由发生变化修改页面title */
  if (to.meta) {
    document.title = to.meta;
  }

  // 未匹配到任何页面, 跳转到404页面
  if (to.matched.length === 0) {
    router.push('/404');
  }
  if (to.path === '/404') {
    next();
    return;
  }

  const token: any = localStorage.getItem('siteToken');
  if (!token) {
    // 非登陆状态
    if (to.path !== '/bind' && to.path !== '/entrust' && to.path !== '/entrustPlan' && to.path !== '/') {
      // 除了登录页 && 在线委托页 && 星级房屋托管计划，其他将跳转到登陆页
      router.push('/bind');
    }
    next();
  }
  next();
});

export default router;

