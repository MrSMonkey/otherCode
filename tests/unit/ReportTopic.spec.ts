/*
 * @Description: ReportTopic组件单元测试
 * @Author: LiuZhen
 * @Date: 2018-10-10 17:35:09
 * @Last Modified by: LiuZhen
 * @Last Modified time: 2019-01-22 18:08:22
 */
import { shallowMount, createLocalVue } from '@vue/test-utils';
import router from 'vue-router';
import Vuex from 'vuex';
import ReportTopic from '@/views/ReportTopic.vue';

// 独立构建Vue构造函数
const localVue = createLocalVue();
localVue.use(router);
localVue.use(Vuex);

describe('对ReportTopic组件进行的单元测试', () => {
  let store: any;
  let wrapper: any;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        global: {}
      }
    });

    // 挂载组件
    wrapper = shallowMount(ReportTopic, { store, localVue });
  });

  afterEach(() => {
    // 销毁实例
    wrapper.destroy();
  });

  test('ReportTopic', () => {
    // 测试 ReportToopic 是一个Vue组件
    expect(wrapper.isVueInstance()).toBeTruthy();
    // 存在ReportTopic组件
    expect(wrapper.is(ReportTopic)).toBeTruthy();
    // 测试节点被正确挂载
    expect(wrapper.find('.theme-bj').exists()).toBeTruthy();
    // 测试数据 routeLists 的类型为数组
    expect(wrapper.vm.routeLists instanceof Array).toBeTruthy();
    // 测试数据 routeLists 的长度不为0
    expect(wrapper.vm.routeLists.length > 0).toBeTruthy();
  });

  // 测试 scrollToTop 方法被调用，页面 scrollTop的值为0
  test('scrollToTop method', () => {
    wrapper.vm.scrollToTop();
    expect(document.documentElement!.scrollTop === 0).toBeTruthy();
  });

  // 生成测试快照
  test('create expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
