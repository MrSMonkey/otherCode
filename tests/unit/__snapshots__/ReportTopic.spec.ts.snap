// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`对ReportTopic组件进行的单元测试 create expected html structure 1`] = `
<div
  class="theme-bj"
>
  <bubble-stub
    title="报表主题"
  />
   
  <div
    class="theme-wrap"
  >
    <ul
      class="theme-list clearfix"
    >
      <router-link-stub
        event="click"
        tag="a"
        to="/businessOpa"
      >
        <li
          class="theme-item"
        >
          <div
            class="li-inner"
          >
            <p
              class="theme-p"
            >
              <i
                class="uk-icon uk-icon-yewu"
              />
            </p>
             
            <p
              class="theme-label"
            >
              业务经营
            </p>
          </div>
        </li>
      </router-link-stub>
      <router-link-stub
        event="click"
        tag="a"
        to="/home"
      >
        <li
          class="theme-item"
        >
          <div
            class="li-inner"
          >
            <p
              class="theme-p"
            >
              <i
                class="uk-icon uk-icon-zhuti"
              />
            </p>
             
            <p
              class="theme-label"
            >
              数据质量
            </p>
          </div>
        </li>
      </router-link-stub>
    </ul>
  </div>
   
  <bottombtn-stub
    btn="reportTopic"
  />
</div>
`;
